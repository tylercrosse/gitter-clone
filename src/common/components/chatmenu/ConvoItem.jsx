import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

const ConvoItem = ({ name, title }) => (
  <div className="convo-item">
    <Link className="convo-item-container" to={'/' + name}>
      <img className="convo-item-avatar" src={'http://i.pravatar.cc/22?u=' + name} alt={name} />
      <h2 className="convo-item-title">{title}</h2>
    </Link>
  </div>
);

ConvoItem.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default ConvoItem;
