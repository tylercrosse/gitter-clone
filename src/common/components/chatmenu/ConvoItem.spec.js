import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ConvoItem from './ConvoItem';

const setup = () => {
  const props = {
    id: '5898b15d6047fd8db8b4e3bb',
    updatedAt: '2017-02-06T17:24:45.657Z',
    createdAt: '2017-02-06T17:24:45.657Z',
    name: 'chat2',
    __v: 0
  };
  const component = <ConvoItem {...props} />;

  return {
    props,
    component
  };
};

describe('<ConvoItem />', () => {
  it('should render correctly', () => {
    const { component } = setup();
    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
