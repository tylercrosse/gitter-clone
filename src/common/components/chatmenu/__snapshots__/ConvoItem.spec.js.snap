// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConvoItem /> should render correctly 1`] = `
<div
  className="convo-item"
>
  <a
    className="convo-item-container"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="chat2"
      className="convo-item-avatar"
      src="http://i.pravatar.cc/22?u=chat2"
    />
    <h2
      className="convo-item-title"
    />
  </a>
</div>
`;
